src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -121,7 +121,23 @@ internal sealed class PdbSourceDocumentMetadataAsSourceFileProvider : IMetadataA
                         .AddReferences(dllReference);
 
                     var key = SymbolKey.Create(symbolToFind, cancellationToken);
-                    var newSymbol = key.Resolve(tmpCompilation, ignoreAssemblyKey: true, cancellationToken).Symbol;
+                    var resolution = key.Resolve(tmpCompilation, ignoreAssemblyKey: true, cancellationToken);
+                    var newSymbol = resolution.Symbol;
+                    if (newSymbol is null && resolution.CandidateReason == CandidateReason.Ambiguous)
+                    {
+                        // If we followed a type forward then there will be an ambiguous resolution between the implementation
+                        // and any references that type forwarded to get to it, so we need to find the symbol from the correct
+                        // assembly.
```

Comment:
This seems kinda strange -- I'd assume the references that forwarded would have just been resolved correctly. Is it because the forwards are broken in some case, or this is actually a correct forward?

Reply:
Or do we do this intentionally in other places because we need other resolution logic there?

Reply:
The issue here is that we end up with a compilation that has a reference to the reference assembly containing type `C`, but with nothing else useful, and an implementation assembly containing type `C` that has actual useful code. The type forward itself is not in play, and the DLL that contained them is not part of any compilation. Having said that, I can see how my wording of "any references that type forwarded to get to it" is confusing, so I will update.